meta-oe/recipes-devtools/perfetto/perfetto.bb

```diff
@@ -0,0 +1,151 @@
+SRCREV_clang-format = "96636aa0e9f047f17447f2d45a094d0b59ed7917"
+
+DEPENDS += " ninja-native"
+DEPENDS += " curl-native"
+
+# To avoid the GNU_HASH warning
+TARGET_CC_ARCH += "${LDFLAGS}"
+
+S = "${WORKDIR}/git"
+
+FILES:${PN}:append = " \
+  ${bindir}/tracebox \
+  "
+
+# Run the GN (Generate Ninja) script using the Linux-Arm configuration
+# and replace the compiler flags
+do_configure () {
+    BUILD_DIR=${WORKDIR}"/build"
+    echo "S=${S}"
+
+    # Rename a few build tools if they have not been renamed
+    cd buildtools
+    x="libbacktrace-177940370e4a6b2509e92a0aaa9749184e64af43"
+    [ -d $x ] && mv $x libbacktrace
+    x="sqlite-amalgamation-3350400"
+    [ -d $x ] && mv $x sqlite
+    x="gn-linux64-1968-0725d782"
+    [ -f $x ] && mkdir linux64 && mv $x linux64/gn
+    chmod +x linux64/gn
+    cd ..
+
+    CC_BIN=`echo $CC | awk '{print $1}'`
+    CXX_BIN=`echo $CXX | awk '{print $1}'`
+    STRIP_BIN=`echo $STRIP | awk '{print $1}'`
+
+    ARGS="is_debug=false "  # Tell gn to use release mode
+
+    if [ -z `echo ${TOOLCHAIN} | grep clang` ]; then
+        ARGS=$ARGS" is_clang=false"
+    else
+        ARGS=$ARGS" is_clang=true"
+    fi
+    ARGS=$ARGS" target_os=\"linux\""
+    ARGS=$ARGS" target_cpu=\"arm\""
+    ARGS=$ARGS" target_cc=\"$CC_BIN\""    # Use Yocto's CC
+    ARGS=$ARGS" target_cxx=\"$CXX_BIN\""  # Use Yocto's CXX
+    ARGS=$ARGS" target_strip=\"$STRIP_BIN\"" #
+    ARGS=$ARGS" target_sysroot=\"${RECIPE_SYSROOT}\""
+    ARGS=$ARGS" target_linker=\"$CC_BIN\""
+    ARGS=$ARGS" target_ar=\"$AR\""  # Use Yocto's AR
+    ARGS="'$ARGS'"
+    cmd="tools/gn gen --args=$ARGS $BUILD_DIR"
+
+    echo $cmd
+    # Use eval, not just call $cmd, due to escaping of single quotation marks
+    eval $cmd
+
+    cd $BUILD_DIR
+    # Eliminate a few incompatible build flags
+    REPLACES="s/-Wl,--icf=all//g"
+    REPLACES=$REPLACES";s/-Werror//g"
+    REPLACES=$REPLACES";s/-mfpu=neon//g"
+    REPLACES=$REPLACES";s/-fcolor-diagnostics//g"
+    REPLACES=$REPLACES";s/=format-security//g"
+    REPLACES=$REPLACES";s/-fdiagnostics-show-template-tree//g"
+    REPLACES=$REPLACES";s/-D_FORTIFY_SOURCE=2//g"
+    REPLACES=$REPLACES";s/-fuse-ld=\S*/-fuse-ld=gold/g"
+    
+    # Eliminate all dependency checks done by install-build-deps
+    REPLACES=$REPLACES";s/python3.*--check-only=\S*/true/g"
+    find . -name "*.ninja" | xargs sed $REPLACES -i
+
+    # Done processing the Ninja files
+}
+
+# Perfetto generates a few different binaries, such as traced and traced_probes and perfetto.
+# The "tracebox" is a busybox that combines the 3 above and provides a single stop for doing the trace capture work, so we only build "tracebox" here.
+do_compile () {
+    cd ${WORKDIR}/build/
+    ninja -C . tracebox
+}
+
+do_install () {
+    BUILD_DIR=${WORKDIR}/build/
+    BIN_DIR=${D}${bindir}
+
+    install -d -m0755 $BIN_DIR
+    install $BUILD_DIR/tracebox $BIN_DIR/tracebox
+}
+
+do_package_qa() {
+    echo "Skipping package_qa to silence the GNU_HASH error"
```

Comment:
Use INSANE_SKIP variable if it's too difficult to pass correct LDFLAGS into the build, no need to skip the whole function.

Reply:
It turns out passing the correct LDFLAGS is pretty easy, so this custom do_package_qa is removed.